### DISCUSSION TITLE
Welcome to PlanZ Discussions!

### DISCUSSION THREAD

**Discussion by LVerhulst4321** [LVerhulst4321/PlanZ#21] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lostcarpark** [comment]
Here's a brief introduction.
My name is James Shields, and I've been attending conventions since 1991 and involved in organisation since 1993. Starting with the Irish national convention, Octocon, I've assisted with various Irish and UK conventions, as well as UK Eastercons, Eurocons and Worldcons.
I have been worked on convention websites since 1996, and built a very basic registration system for Octocon in 2000, which was rewritten a number of times over the years. I started a new registration system ConReg, build on Drupal 8, which has been in use by Octocon since 2016, and has been used by Eastercons, and a number of other conventions.
I set up Zambia for the 2021 Eastercon (at fairly short notice), which led to me contributing a number of changes to that project.
I also started the ConClár project at the start of 2022, and it has already received over 200 commits and over 80 pull requests, and been used by 2 conventions.
I've a number of things I'd like to see develop in PlanZ. In the short term, I would like to have a links section as part of participant profiles. I would also to finish the participant workflow I started for Zambia.
Longer term, I would like to see a more modular approach taken, with support for installable modules, so that features don't have to be baked into the core project. This would be particularly useful for features that aren't needed by all conventions, and would help avoid having multiple branches for different conventions.

**Reply by bcholmes** [reply]
I agree with the desire for a links section. Folks often want to hype their web page, their Twitter account, their Faecebook, or whatever. Many other sites give easily clickable icons or buttons to allow folks to follow an engaging speaking/panelist on their favourite social media. I think there are gonna be interesting policy questions, mind you. Should the tool be neutral about, say, letting someone link to their OnlyFans?

I also agree with the desire for some kind of architecture or api to allow for drop-in modules.

**Reply by lostcarpark** [reply]
I've commented on the links part on that discussion.

One think PHP is very good at is discoverable code. Making it walk through a folder structure and find modules isn't very hard. Of course we would need to define endpoints to allow the modules to interact with the core functionality, and that would by no means be trivial. But I think there are plenty of areas we could start small, such as reports.

**Reply by bcholmes** [reply]
It's probably a fraught example (what with the barrage of hacking attempts on Our Fine Internet), but WordPress is undoubtedly the exemplar for PHP plug-in architecture.

**Reply by LVerhulst4321** [reply]
Woo! That's some big changes you both are discussing! Exciting. 
I hope I can help, but my coding skills are a little lacking in this new world. I don't really know java (any flavor) at all. But, willing to learn as time permits. :-)

**Comment by bcholmes** [comment]
I'm BC. By day, I'm a computer nerd. But by night, I'm still a computer nerd, but one who uses her nerdish powers to work on [WisCon](https://wiscon.net/) app development. I've been building/enhancing/deploying their mobile [scheduling](https://play.google.com/store/apps/details?id=org.ayizan.android.wissched) [app](https://apps.apple.com/ca/app/wissched/id524251160) for about 10 years (the app also doubles as the FogCon scheduling app). In the last year, I've deployed a FrankenZambia to replace WisCon's aging (but quite extensive) programming system. 

This year, I also built the [reg system](https://reg.wiscon.net/) for WisCon (that can interact with my FrankenZambia).

Other things I tend to care about:

- Good user experience (especially for people who don't think like computer nerds)
- Privacy technologies
- Visual information
- Mobile

**Comment by LVerhulst4321** [comment]
Brief intro. 
My name is Leane Verhulst. I have been attending conventions since the late 90s and got involved in helping at the local level in the early 2000s. The job that I got involved in had to do with scheduling - helping the various parts of the convention not double book rooms or people and making sure that all of that information was compiled together for publications and other areas. This led to me writing some software to help with this task. 
While I am a programmer in my day job, I am not a web programmer, so my first bit of code only lived locally on my computer. That worked for a couple of years, but I wanted to make something web-based so it wasn't all relying on me. I didn't want to be a bottleneck. So I wrote a package called ConCentric - Event Planning Software and put it on SourceForge. It wasn't the best - mostly because I was learning PHP bit by bit as I wrote it and so it had a lot of quirks. And it wasn't that easy to update. But it worked for us for a number of years and a couple other conventions used it as well.
Eventually, there were enough requests for new features, that I looked around and found Zambia that had some of the new features that was being requested. I downloaded the GitHub package and have been working with it since at least 2016. I've made a bunch of changes over the years, some of which were incorporated into the original base code, some not, some that fell away as not useful.
Lately, the base code, the main branch of Zambia, isn't being updated very much. So, I decided to do a big fork to make a lot of changes and invited a few others to help. Hopefully, more will help and this software will keep growing and evolving.

Welcome to PlanZ.
